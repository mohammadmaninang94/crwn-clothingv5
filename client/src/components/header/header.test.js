
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from 'redux-mock-store';
import { MemoryRouter } from "react-router-dom";
import renderer from 'react-test-renderer';

import Header from './header.component';

describe('Header rendering', () => {
    const mockStore = configureStore([]);
    const wrapper = (store) => {
        return (<Provider store={store}>
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        </Provider>)
    };
    const renderComponent = (state) => {
        const store = mockStore(state);
        return [
            render(wrapper(store)),
            store,
        ]
    };

    test('initial render should display sign in button and cart 0', () => {
        const [{ getByRole, queryByRole, queryByText }, store] = renderComponent({
            user: {
                currentUser: null,
                signInMessage: null,
                signUpMessage: null,
                signOutMessage: null,
            },
            cart: {
                hidden: true,
                sidebar: true,
                items: []
            }
        });

        // console.log(queryByText(store.getState().cart.items.length.toString()));
        // console.log(store.getState().cart.items.length.toString());

        expect(getByRole('link', { name: /sign in/i })).toBeInTheDocument();
        expect(queryByRole('link', { name: /sign out/i })).not.toBeInTheDocument();
        expect(queryByText(store.getState().cart.items.length.toString())).toBeInTheDocument();


        const tree = renderer
            .create(wrapper(store))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('initial render should display sign out button and with cart items count', () => {
        const [{ queryByText }, store] = renderComponent({
            user: {
                currentUser: {},
                signInMessage: null,
                signUpMessage: null,
                signOutMessage: null,
            },
            cart: {
                hidden: true,
                sidebar: true,
                items: [{
                    imageUrl: '/images/shop-img/jackets/black-shearling.png',
                    name: 'Black Jean Shearling',
                    id: 18, price: 125,
                    quantity: 1
                }]
            }
        });

        expect(queryByText(/sign in/i)).not.toBeInTheDocument();
        expect(queryByText(/sign out/i)).toBeInTheDocument();
        expect(queryByText(store.getState().cart.items.length.toString())).toBeInTheDocument();

        const tree = renderer
            .create(wrapper(store))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});