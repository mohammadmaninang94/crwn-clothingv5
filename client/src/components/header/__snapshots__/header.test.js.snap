// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header rendering initial render should display sign in button and cart 0 1`] = `
<div
  className="sc-hKgILt edpTQa"
>
  <a
    aria-current="page"
    className="sc-eCssSg fRGvgp header__link--active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <svg>
      crown.svg
    </svg>
  </a>
  <nav
    className="sc-jSgupP fNIVQT"
  >
    <a
      aria-current={null}
      className="sc-gKsewC dllOHe"
      href="/shop"
      onClick={[Function]}
    >
      shop
    </a>
    <a
      aria-current={null}
      className="sc-gKsewC dllOHe"
      href="/contact"
      onClick={[Function]}
    >
      contact
    </a>
    <a
      aria-current={null}
      className="sc-gKsewC dllOHe"
      href="/signin"
      onClick={[Function]}
    >
      sign in
    </a>
    <div
      className="sc-bdfBwQ yXoxa"
      onClick={[Function]}
    >
      <svg
        className="sc-gsTCUz iapUUP"
      >
        shopping-bag.svg
      </svg>
      <span
        className="sc-dlfnbm hmRxPi"
      >
        0
      </span>
    </div>
  </nav>
</div>
`;

exports[`Header rendering initial render should display sign out button and with cart items count 1`] = `
<div
  className="sc-hKgILt edpTQa"
>
  <a
    aria-current="page"
    className="sc-eCssSg fRGvgp header__link--active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    <svg>
      crown.svg
    </svg>
  </a>
  <nav
    className="sc-jSgupP fNIVQT"
  >
    <a
      aria-current={null}
      className="sc-gKsewC dllOHe"
      href="/shop"
      onClick={[Function]}
    >
      shop
    </a>
    <a
      aria-current={null}
      className="sc-gKsewC dllOHe"
      href="/contact"
      onClick={[Function]}
    >
      contact
    </a>
    <div
      className="sc-gKsewC dllOHe"
      onClick={[Function]}
    >
      sign out
    </div>
    <div
      className="sc-bdfBwQ yXoxa"
      onClick={[Function]}
    >
      <svg
        className="sc-gsTCUz iapUUP"
      >
        shopping-bag.svg
      </svg>
      <span
        className="sc-dlfnbm hmRxPi"
      >
        1
      </span>
    </div>
  </nav>
</div>
`;
